### ISSUE TITLE
How to conver the PyTorch Data Loader to SV Detection Object

### ISSUE THREAD

**Issue by shanalikhan** [roboflow/supervision#1683]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

I've Dataloader object and I am able to convert the predicted model output using `from_transformer` to Detection Object. Since i want to generate the MAP and confusion matrix, I need to convert the actual object to Detection object as well. 

How to do that?

```
for idx,batch in enumerate(tqdm(TEST_DATALOADER)):
    
    pixel_values = batch["pixel_values"].to(DEVICE)
    pixel_mask = batch["pixel_mask"].to(DEVICE)
    #print(batch["labels"])
    labels = [{k: v.to(DEVICE) for k, v in t.items()} for t in batch["labels"]]
    #print(labels)
    with torch.no_grad(): 
      outputs = model(pixel_values=pixel_values, pixel_mask=pixel_mask)
      #outputs = model(**batch)

    orig_target_sizes = torch.stack([target["orig_size"] for target in labels], dim=0)
    #print(outputs)
    results = image_processor.post_process_object_detection(outputs, target_sizes=orig_target_sizes,threshold=0.5)
    print("LABELS")
    print(labels[0].keys())
    
    print(labels[0]['class_labels'])
    print("RESULTS")
    print(results[0].keys())
    print(results[0]['labels'])
    for result in results:
      detections = sv.Detections.from_transformers(result)#.with_nms(threshold=IOU_TRESHOLD)
      #sv.Detections.from_
      targets.append(annotations)
      predictions.append(detections)
    break
```

**Comment by Kanishk-tiwari-045** [comment]
Hi @shanalikhan thanks for the question!

## To generate evaluation metrics like MAP and the confusion matrix, you just need to convert your ground truth labels to sv.Detections as well.

### ✅ Convert Ground Truth Labels (Targets) to sv.Detections:
```py
for label in labels:  # labels per batch
    boxes = label["boxes"].cpu().numpy()                # shape: [N, 4], assumed to be in xyxy format
    class_ids = label["class_labels"].cpu().numpy()     # shape: [N]
    target_detections = sv.Detections(
        xyxy=boxes,
        class_id=class_ids
    )
    targets.append(target_detections)
```
### ✅ Convert boxes in xywh to xyxy format like this:
```py
from supervision.utils.bbox import xywh_to_xyxy

boxes = xywh_to_xyxy(label["boxes"].cpu().numpy())
```
Once both targets and predictions are lists of sv.Detections, you can compute metrics:

### ✅ Evaluate with MAP and Confusion Matrix

### MAP
```py
map_metric = MeanAveragePrecision()
map_metric.update(predictions, targets)
print(map_metric.compute().map50_95)

### Confusion Matrix
conf_matrix = ConfusionMatrix.from_detections(
    predictions=predictions,
    targets=targets,
    classes=["class_1", "class_2", "class_3"]  # replace with your class names
)
print(conf_matrix.matrix)
```
I can help adapting to these changes, please assign me the task to do so.

Hope this helps!

**Comment by salmanwnl44** [comment]
is it open? Can I work on this?

**Comment by pateljeffery79545** [comment]
刚 star，准备周末复现一遍 💡

**Comment by Borda** [comment]
> is it open? Can I work on this?

@salmanwnl44, sure, go ahead :)
